Refactor `widgetsDisabled` & `showToolbar` to props

mayagbarnes opened the pull request:
## Describe your changes
**Part 6 of `StreamlitContextProvider` updates**

This PR refactors `widgetsDisabled` &  `showToolbar` to props and removes the now empty `AppContext`.
* `showToolbar`: only passed a couple levels to its single consumer (`Header`) so doesn't make sense to use in a context
* `widgetsDisabled`: performance analysis demonstrated that passing this as prop vs. a dedicated context was equivalent (did not prevent intermediate component re-renders). Not adding to existing `ScriptRunContext` as it mixes concerns with unrelated properties.

This PR also flattens the `libConfig` values passed to `StreamlitContextProvider` for future context work.

## Testing Plan
- JS Unit Tests: ✅ Updated

mayagbarnes committed:
Refactor widgetsDisabled and showToolbar to props

mayagbarnes committed:
Flatten libConfig properties

mayagbarnes commented:
* **#12896** <a href="https://app.graphite.dev/github/pr/streamlit/streamlit/12896?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12895** <a href="https://app.graphite.dev/github/pr/streamlit/streamlit/12895?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/streamlit/streamlit/12895?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#12884** <a href="https://app.graphite.dev/github/pr/streamlit/streamlit/12884?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12791** <a href="https://app.graphite.dev/github/pr/streamlit/streamlit/12791?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12790** <a href="https://app.graphite.dev/github/pr/streamlit/streamlit/12790?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12789** <a href="https://app.graphite.dev/github/pr/streamlit/streamlit/12789?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#12788** <a href="https://app.graphite.dev/github/pr/streamlit/streamlit/12788?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `develop`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

mayagbarnes committed:
Fix test

sfc-gh-bnisco commented on frontend/app/src/components/AppView/AppView.test.tsx:
```diff
@@ -1218,20 +1162,11 @@ describe("AppView element", () => {
     })
 
     it("header shows toolbar actions in embed mode WITH show_toolbar", () => {
-      // Mock embed mode with show_toolbar (showToolbar = true)
```
The simplification here is great!

sfc-gh-bnisco commented on frontend/lib/src/hooks/useCrossOriginAttribute.test.tsx:
```diff
@@ -33,13 +35,20 @@ const getWrapper = (
-        }) as unknown as LibContextProps,
-      []
-    )
+    // useMemo is not needed here as there is no performance benefit
```
**note:** This comment makes sense, it might be worth thinking about not applying the `@eslint-react/no-unstable-context-value` rule in test files, but that is not in scope for this PR.

sfc-gh-bnisco commented on frontend/lib/src/index.ts:
```diff
@@ -30,7 +30,9 @@ export type { FormsContextProps } from "./components/core/FormsContext"
 export { LibContext } from "./components/core/LibContext"
-export type { LibConfig, LibContextProps } from "./components/core/LibContext"
+export type { LibContextProps } from "./components/core/LibContext"
+// Re-export LibConfig from connection package for convenience
```
**nit:** Consider removing this comment as it doesn't provide any extra insight into what the line of code is doing.

mayagbarnes committed:
Remove unnecessary comment

kmcgrady committed:
Create RenderNodeVisitor (#12814)

Now, we introduce a new visitor for rendering the block react node. We
made this separate from the render-tree folder because we did not want
to make React a dependency. This is great because we can break out the
render-tree to a separate package someday.

- JS Unit Tests for the RenderNode Visitor
- E2E tests should ensure the rendering logic remains correct

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this
project are made under the Apache 2.0 license.

mayagbarnes committed:
Fix
